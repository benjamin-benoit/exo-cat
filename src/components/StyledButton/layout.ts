import styled from 'styled-components';

export const CustomButton = styled.button`
  background-color: #8FD6BD;
  border: none;
  padding: 10px 40px 10px;
  cursor: pointer;
  border-radius: 10px;
`;
