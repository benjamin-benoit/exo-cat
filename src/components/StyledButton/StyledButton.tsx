import React from 'react';
import { CustomButton } from './layout';

interface Props {
  text: string;
  onClick: () => void;
}

const StyledButton = ({ text, onClick }: Props) => {
  return <CustomButton onClick={onClick}>{text}</CustomButton>;
};

export default StyledButton;
