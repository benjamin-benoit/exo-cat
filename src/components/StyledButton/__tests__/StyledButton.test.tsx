import React from 'react';
import userEvent from '@testing-library/user-event';
import StyledButton from '../StyledButton';
import renderer from 'react-test-renderer';

describe('Test Button component', () => {
  let container: any = null;
  let clickFn: any = null;

  beforeEach(() => {
    clickFn = jest.fn();
    container = renderer.create(<StyledButton onClick={clickFn} text="send" />);
  });

  it('button Clicked', () => {
    userEvent.click(container);
    expect(clickFn).toHaveBeenCalled();
  });
});
