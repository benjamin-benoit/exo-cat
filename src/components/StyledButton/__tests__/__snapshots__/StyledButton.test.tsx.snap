// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button click work correctly 1`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;

exports[`button correctly 1`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;

exports[`renders button correctly 1`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;

exports[`renders button correctly 2`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;

exports[`renders card correctly 1`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;

exports[`test button click 1`] = `
<button
  className="sc-bcXHqe bcTVIz"
  onClick={[Function]}
>
  test
</button>
`;
